qiskit/aqua/operators/list_ops/summed_op.py

```diff
@@ -142,3 +143,20 @@ def to_legacy_op(self, massive: bool = False) -> LegacyBaseOperator:
             coeff = cast(float, self.coeff)
 
         return self.combo_fn(legacy_ops) * coeff
+
+    def chop_summands(self, threshold = 0) -> 'SummedOp':
```

Comment:
Take a look at the type for coeff in the constructor say of PrimitiveOp and you will see this `Union[int, float, complex, ParameterExpression]` as its type. Ie it can be any one of those types, an integer, a float etc. Clearly only when it is of type complex will it have imaginary and complex parts. When its a ParameterExpression maybe there is no chopping that (this is a symbolic placeholder which will be substituted by a actual value later). If the expression would always evaluate to 0 no matter what was subsituted, I could see it being chopped, but it seems like an expression you would never really do in practice. Hence accounting for this weird case seems to much.

Also bear in mind that oplist can contain OperatorBase entities that themselves are lists so things need to cater to such a nested structure.

Also should all the coeffs be below the theshold and all get removed by the chop I do not know whether its valid to have an empty oplist. One thing we do in Aqua where we had an WeightedPauliOperator that was empty was to create this in its stead for opflow https://github.com/Qiskit/qiskit-aqua/blob/146bff86d486dfd73a871e7f3c000015bc872581/qiskit/aqua/algorithms/minimum_eigen_solvers/vqe.py#L234
So maybe knowing the number of qubits on the opertor being chopped, if it were to end up in an empty oplist, then the oplist shoud be modified to contain at least this. I am not sure its a great solution but at this stage I am not sure how to do better.

Reply:
Just to make sure I'm following here is my plan:

1. When iterating through the oplist check each type
2. If it's int or float just compare it with the threshold
3. If it's complex compare the real and imaginary components against the threshold
4. If it's ParameterExpression leave it as-is
5. If the new oplist is empty, put a zero_op in it

The one thing I'm still unclear about it are OperatorBase entities. It looks like it could be a ListOp, so would I just handle that in my type check? 

Lastly:
Initially I was assuming the coefficients were real numbers but then Julien told me on Slack `make sure that you’re comparing the real and imaginary part separately, because the coefficients are complex numbers.` which led me to believe this function would only be receiving complex numbers. It sounds like this is not the case?

Reply:
You only have to look at PrimitiveOp init which has coeff as a Union (ie one of) a list of different types and see that it is stored in the instance just as it is passed in with no type change at all. Indeed look at the getter there of coeff - it says what you will get back is a Union of that same set of types.

In regards of operators in the oplist that are based on ListOp, yes the should be handled recursively. Normally the operations we have done before have had the method done on all types so the type itself handle the operation. The chop  so far you are doing just on SummedOp it seems so it could not be done recursively in this case (except if the SummedOp contains one or more other SummedOps in its list). Bear in mind too that these ListOp based operators have their own coeff as a factor for the entire list - each operator in the list having its own too (whether a primitive op or another list op based one). You can see simplify here combines the coeffs so that the coeff of the resultant simplified SummedOp will be 1 i.e. each operator in the list had its own coeff multiplied with the SummedOp coeff so the net result is equivalent.

Hope this helps.